assets/js/base/hooks/shipping/use-shipping-address.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { useEffect, useReducer } from '@wordpress/element';
+import isShallowEqual from '@wordpress/is-shallow-equal';
+import { useDebounce } from 'use-debounce';
+import { CART_STORE_KEY as storeKey } from '@woocommerce/block-data';
+
+/**
+ * Internal dependencies
+ */
+import { useStoreCart } from '../cart/use-store-cart';
+import { pluckAddress } from '../../utils';
+
+const reducer = ( state, address ) =>
+	isShallowEqual( pluckAddress( address ), pluckAddress( state ) )
+		? state
+		: { ...state, ...address };
```

Comment:
I moved this logic into the reducer and out of the effect because this simplifies the hook implementation. With this change `debouncedShippingAddress` only updates when the `useReducer` state updates and that state will only update when there is a difference between the incoming pluckedAddress and the current address in the state (rather than change the state everytime the reducer was invoked).

Reply:
`useReducer` also controls the local state of that form (as well as cart local state), by having this there 
`isShallowEqual( pluckAddress( address ), pluckAddress( state ) )` we refuse all state changes that are not related to shipping fields (country, state, city, postcode).

the goal of that line was to simply cancel API actions if those fields didn’t change

Reply:
Ya good catch, this will have more importance in the checkout when there are more fields that could change, so good call there.